### ISSUE TITLE
Error 403 when changing user password

### ISSUE THREAD

**Issue by dylanbob** [FreshRSS/FreshRSS#1841]
Hi
I just installed FreshRSS Version 1.10.2 on : Debian 9 using up to date MySQL, php, Apache as a webserver in http, nginx as a reverse proxy in https. I use the auth form to login.
Problem :
Changing my user password (currently connected and admin) on the profile configuration page /i/?c=user&a=profile return a 403 error and disconnect yourself from the service. The password gets changed though.

It is a bit misleading. If you change your password while having deactivated the login, you do not get the error and get a success message like "Your password has been succesfully updated"

I was personnaly trying to set an API password, but firefox automatically filled the first password field (to change the user password) while I wrote my API password on the second field
![Trying to change the API password but the browser automatically filled the first password field](https://user-images.githubusercontent.com/25272703/37683255-ea71b482-2c83-11e8-993f-9481c769a974.png)
![get the error message](https://user-images.githubusercontent.com/25272703/37683271-fbc8dce2-2c83-11e8-97d6-3c643611f679.png)
Nginx and apache logs do not get any useful information. FreshRSS logs do not show anything.

**Comment by Alkarex** [comment]
Hum, we could consider not killing the session after a password change, but if so, it needs to be done carefully. In particular, we still want to kill other potentially open sessions with the same account

**Comment by pled** [comment]
Hum, I had a similar issue with Firefox (59 on Ubuntu) : while trying to set an API password (for EasyRSS use), the first password field (for form) was already filled. Setting the API pwd results in being disconnected (error 403 - Forbidden), and API pwd was not set. Even clearing the first pwd before failed as well. Only way to make it works was using Chrome. Strange !

**Comment by Alkarex** [comment]
@pled I will implement the `autocomplete="new-password"` approach suggested by https://developer.mozilla.org/en-US/docs/Web/Security/Securing_your_site/Turning_off_form_autocompletion

**Comment by nikoss** [comment]
Alive and kicking in...
   Android 7.0
   Freshrss 1.11.2
   Firefox 62.0.3
   Lightning 4.5.1 (browser) 

Of course, I tested it using the desktop web interface of FreshRSS.

It killed the sessions in both browsers. That is, when I tried it in Lightning, I got logged out. Then I switched to Firefox to find out that I was logged out in there, as well! 

Maybe the message could be more clear. Something like...
   "
    Session killed
    Error 403- blah blah
    You don't have blah blah...
   "
Because error 403 makes you think that you don't have access to some file (wrong permissions), or some process doesn't have access rights. But in reality, you just got kicked out. 


The Freshrss logs say...
  " 
Password mismatch for user=USERNAME, nonce=SOME_LETTER-NUMBER_SEQUENCE, c=SOME_LETTER-NUMBER_SEQUENCE

"

I was trying to set an api password.
Whether letting Firefox fill the _login password_, or whether I typed it myself makes no difference.

There is always the chance that I had set a different api password in the past.

**Comment by Alkarex** [comment]
@nikoss In terms of access tokens, all existing Web sessions are killed when you change the password, even when setting the same one again. Yes, we can try to work on the message to make it easier to understand (but I believe an error 403 is the appropriate error code in this case).

If you were setting the API password when this occurred, it can be that an overzealous password manager has auto-filled the other password field (we do have some code to avoid that, but some password managers keep getting more and more aggressive). Could you please try again with another browser that does not have a password manager / form autocomplete? If this is the case, we might have to take even stronger measures to prevent their misbehaving.

So to summarize, as a far as I can see, there is no bug on FreshRSS side, but:
* We should try to make the error message easier to understand
* We might have to implement stronger measures to prevent misbehaving password managers, if it turns out that this is again the problem.

**Comment by nikoss** [comment]
I tried with another browser (desktop computer) and Firefox in mobile again, and it is as you say. 

When the login password is filled (whether by me or by the browser) the session gets killed. 

If I delete the login password (or it is blank), and just type the api password, the session is not terminated. I get a pop up message, something about "your profile was updated".

So not, a bug after all. But a little bit confusing, though. "Should I type the login password and the api password?" "Should I just type the api password?" "Ha, just type them both, to be sure!!! "
